src/cryptography/hazmat/backends/openssl/ciphers.py

```diff
@@ -149,7 +149,7 @@ def update(self, data: bytes) -> bytes:
         n = self.update_into(data, buf)
         return bytes(buf[:n])
 
-    def update_into(self, data: bytes, buf) -> int:
+    def update_into(self, data: bytes, buf: bytes) -> int:
```

Comment:
`bytes` isn't really correct here, why does mypy even allow it?

Reply:
https://docs.python.org/3/library/typing.html#typing.ByteString

```
This type represents the types bytes, bytearray, and memoryview of byte sequences.

As a shorthand for this type, bytes can be used to annotate arguments of any of the types mentioned above.
```

Reply:
It's not correct to pass any of those types though, only a subset of them!

Reply:
Which we enforce at runtime. mypy doesn’t distinguish in a useful manner for us though 😔

Reply:
Right, but the point of typing is indicate invalid types are compile time. Is there really no way in mypy to express "a writable buffer type"!?

Reply:
Sounds like somebody volunteered to determine the answer to that once and for all.